xbmc/platform/win32/filesystem/Win32Directory.cpp

```diff
@@ -81,35 +102,34 @@ bool CWin32Directory::GetDirectory(const CURL& url, CFileItemList &items)
       continue;
     }
 
-    CFileItemPtr pItem(new CFileItem(itemName));
-
-    pItem->SetFolder((findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0);
-    if (pItem->IsFolder())
-      pItem->SetPath(pathWithSlash + itemName + '\\');
-    else
-      pItem->SetPath(pathWithSlash + itemName);
+    const bool isDir = (findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
+    const bool isHidden =
+        (findData.dwFileAttributes & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)) != 0 ||
+        itemName.front() == '.'; // mark files starting from dot as hidden
 
-    if ((findData.dwFileAttributes & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)) != 0
-          || itemName.front() == '.') // mark files starting from dot as hidden
-      pItem->SetProperty("file:hidden", true);
+    std::string itemPath(pathWithSlash + itemName);
+    if (isDir)
+      itemPath += '\\';
 
     // calculation of size and date costs a little on win32
     // so DIR_FLAG_NO_FILE_INFO flag is ignored
-    FileTime localTime{};
-    if (FileTimeToLocalFileTime(reinterpret_cast<FileTime*>(&findData.ftLastWriteTime),
-                                &localTime) == TRUE)
-      pItem->SetDateTime(localTime);
-    else
-      pItem->SetDateTime(0);
+    auto& item = fileItems.emplace_back(std::make_shared<CFileItem>(itemName));
 
-    if (!pItem->IsFolder())
-      pItem->SetSize((__int64(findData.nFileSizeHigh) << 32) + findData.nFileSizeLow);
+    item->SetFolder(isDir);
+    item->SetPath(itemPath);
 
-    items.Add(pItem);
+    if (auto entryTime = GetEntryTime(findData); entryTime.has_value())
```

Comment:
As now time conversion is only one line of code not makes much sense create aux. function `GetEntryTime` to use only here, also `std::optional` cause a little overhead. Why not:

```c++
    FileTime localTime{};
    if (FileTimeToLocalFileTime(reinterpret_cast<const FileTime*>(&findData.ftLastWriteTime),
                                &localTime))
      item->SetDateTime(localTime); 
```